runtime/server/controller.go

```diff
@@ -44,7 +48,20 @@ func (s *Server) ListResources(ctx context.Context, req *runtimev1.ListResources
 		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
-	// TODO: Enforce security policy.
+	for i := 0; i < len(rs); i++ {
+		r := rs[i]
+		r, access, err := s.applySecurityPolicy(ctx, req.InstanceId, r)
+		if err != nil {
+			return nil, status.Error(codes.InvalidArgument, err.Error())
+		}
+		if !access {
+			// Remove from the slice
+			rs[i] = rs[len(rs)-1]
+			rs[len(rs)-1] = nil
+			rs = rs[:len(rs)-1]
+		}
+		rs[i] = r
```

Comment:
This looks odd, as it puts back the resource in it was removed because of no access.

Reply:
Good cathc, it's a bug, fixed it